File: historyscan-back/src/main/java/io/sch/historyscan/infrastructure/config/AppConfig.java
package io.sch.historyscan.infrastructure.config;

import org.springframework.cache.CacheManager;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.scheduling.annotation.AsyncConfigurer;

import java.time.Clock;

@Configuration
public class AppConfig implements AsyncConfigurer {

    @Bean
    public Clock clock() {
        return Clock.systemDefaultZone();
    }

    @Bean
    public CacheManager cacheManager() {
        return new org.springframework.cache.concurrent.ConcurrentMapCacheManager();
    }
}

File: historyscan-back/src/main/kotlin/io/sch/historyscan/infrastructure/config/AppConfig.kt
package io.sch.historyscan.infrastructure.config

import org.springframework.cache.CacheManager
import org.springframework.cache.concurrent.ConcurrentMapCacheManager
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.scheduling.annotation.AsyncConfigurer
import java.time.Clock

@Configuration
class AppConfig : AsyncConfigurer {
    @Bean
    fun clock(): Clock {
        return Clock.systemDefaultZone()
    }

    @Bean
    fun cacheManager(): CacheManager {
        return ConcurrentMapCacheManager()
    }
}
